pkg/dyninst/decode/decoder.go

```diff
@@ -206,35 +206,17 @@ func (d *Decoder) encodeValue(
 			return errors.New("passed data not long enough for pointer")
 		}
 		addr := binary.NativeEndian.Uint64(data)
-		if addr == 0 { // nil pointers are encoded as 0.
-			err := enc.WriteToken(jsontext.String("nil"))
-			if err != nil {
-				return err
-			}
-			return nil
-		}
 		key := typeAndAddr{
 			irType: uint32(v.Pointee.GetID()),
 			addr:   addr,
 		}
 
-		var (
-			header         *output.DataItemHeader
-			pointedData    []byte
-			dontParseValue bool
-		)
+		var address uint64
```

Comment:
We should handle the 0 address case explicitly. It means that the pointer was null. I think that it should not have an address field and it should have `"isNull": true`, it should have the type name and then that's it.